sql/analyzer/pushdown.go

```diff
@@ -16,81 +16,49 @@ func pushdownFilters(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope) (s
 	span, ctx := ctx.Span("pushdown_filters")
 	defer span.Finish()
 
-	if !canDoPushdown(n, scope, a) {
+	if !canDoPushdown(n, a) {
 		return n, nil
 	}
 
-	// First step is to find all col exprs and group them by the table they mention.
-	// Even if they appear multiple times, only the first one will be used.
-	filtersByTable, err := getFiltersByTable(ctx, n)
-
-	// An error returned by getFiltersByTable means that we can't cleanly separate all the filters into tables.
-	// In that case, skip pushing down the filters.
-	// TODO: we could also handle this by keeping track of the filters we can't handle and re-applying them at the end
-	if err != nil {
-		return n, nil
-	}
-
-	indexes, err := getIndexesByTable(ctx, a, n)
+	indexes, err := getIndexesByTable(ctx, a, n, scope)
 	if err != nil {
 		return nil, err
 	}
 
 	exprAliases := getExpressionAliases(n)
-	tableAliases, err := getTableAliases(n)
+	tableAliases, err := getTableAliases(n, scope)
 	if err != nil {
 		return nil, err
 	}
 
-	filters := newFilterSet(filtersByTable, exprAliases, tableAliases)
-
-	n, err = convertFiltersToIndexedAccess(a, n, filters, indexes)
+	n, err = convertFiltersToIndexedAccess(a, n, scope, indexes)
 	if err != nil {
 		return nil, err
 	}
 
-	return transformPushdownFilters(a, n, filters, exprAliases, tableAliases)
+	return transformPushdownFilters(a, n, scope, exprAliases, tableAliases)
 }
 
 // pushdownProjections attempts to push projections down to individual tables that implement sql.ProjectTable
 func pushdownProjections(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope) (sql.Node, error) {
 	span, ctx := ctx.Span("pushdown_projections")
 	defer span.Finish()
 
-	if !canDoPushdown(n, scope, a) {
+	if !canDoPushdown(n, a) {
 		return n, nil
 	}
-	if !canProject(n) {
+	if !canProject(n, a) {
 		return n, nil
 	}
 
-	return transformPushdownProjections(ctx, a, n)
+	return transformPushdownProjections(ctx, a, n, scope)
 }
 
-func canProject(n sql.Node) bool {
+func canProject(n sql.Node, a *Analyzer) bool {
```

Comment:
Just curious why we threaded this here and in `canDoPushdown`...

Reply:
For logging. But it was unused in one of those places, removed.